packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -158,10 +161,21 @@ export const useGridKeyboardNavigation = (
   );
 
   const handleColumnHeaderKeyDown = React.useCallback<
-    GridEventListener<GridEvents.columnHeaderNavigationKeyDown>
+    GridEventListener<GridEvents.columnHeaderKeyDown>
   >(
     (params, event) => {
-      event.preventDefault();
+      if (
+        !isGridHeaderCellRoot(event.target as HTMLElement) &&
+        !(params.field === GRID_CHECKBOX_SELECTION_COL_DEF.field && isNavigationKey(event.key))
+      ) {
+        // When focus is on a nested input, keyboard events have no effect to avoid conflicts with native events.
+        // There is on exception for the checkBoxHeader
+        return;
+      }
+
+      if (isNavigationKey(event.key)) {
+        event.preventDefault();
+      }
       if (!params.field) {
         return;
       }
```

Comment:
To improve readability.

```suggestion

      if (!params.field) {
        return;
      }

```

Reply:
This condition is useless I added it to apply this PR comment but I miss understood it 😅 https://github.com/mui-org/material-ui-x/pull/3275#discussion_r758851911